modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1664,6 +1665,38 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
   protected abstract getAddressInfoFromExplorer(address: string, apiKey?: string): Bluebird<AddressInfo>;
   protected abstract getUnspentInfoFromExplorer(address: string, apiKey?: string): Bluebird<UnspentInfo[]>;
 
+
+  /**
+   * Derive child keys based on bip32 paths (must start with "m")
+   * If no userKeyPath is provided, all derivation of three keys will use the path parameter
+   * Else, user key is derived with the userKeyPath, and backup and bitgo keys still use path.
+   * @param {bitcoin.HDNode[]} keyArray
+   * @param {string} path
+   * @param {string} userKeyPath
+   * @returns {bitcoin.HDNode[]}
+   */
+   deriveKeysByPaths(keyArray: bitcoin.HDNode[], path: string, userKeyPath?: string): bitcoin.HDNode[] {
+    if (userKeyPath) {
+      const derivedKeys: bitcoin.HDNode[] = [];
+      const userKey = keyArray.shift();
+      derivedKeys.push(userKey.derivePath(userKeyPath));
+      const keys = keyArray.map((k) => k.derivePath(path));
+      derivedKeys.concat(keys);
+      return derivedKeys;
+    }
+    return keyArray.map((k) => k.derivePath(path));
+  }
+
+  /**
+   * Derive child keys at specific index, from provided parent keys
+   * @param {bitcoin.HDNode[]} keyArray
+   * @param {number} index
+   * @returns {Ed25519PrivateKey[]}
```

Comment:
Is there something generic about the `Ed25519PrivateKey` object? Because I think that's the wrong curve for Bitcoin, but I might be misunderstanding the meaning of "ed25519" or "bitcoin.HDNode" in this context.

Reply:
should be `bitcoin.HDNode[]`